spoon-runner/src/main/java/com/squareup/spoon/CliArgs.kt

```diff
@@ -70,6 +74,15 @@ internal class CliArgs(parser: ArgParser) {
 
   val coverage by parser.flagging("Enable code coverage")
 
+  private fun validateInstrumentationArgs() {
+    val isTestRunPackageLimited = instrumentationArgs?.contains("package") ?: false
```

Comment:
since you can pass `class-name` as an argument to spoon but there is no official `package` argument this package-specific code seems out of place to me.

would rather see a new `package` argument, think that would be really useful

Reply:
Dedicated package arg can be added but it will be only 'shortcut' to instrumentation arg as class-name is. Validation uses build in arg parser feature - imho it's better to add some logic here instead of having strange behaviours later. For example current master version allows to pass class (or class-name) and package but few things will happen:
1. Runner will gather test to execute list based on package.
1. Runner will try to run every test
1. Every single test will silently fail because internally am instrument doesn't allow both class and package.

Reply:
Yes it is only a shortcut but at least then it would be better documented as an option.  I didn't even realize you could use package like this and I've been using spoon awhile, so thanks for the info 😄 

I totally understand the purpose of the validation.  IMO it is a code smell to be doing validation on instrumentationArgs for package when package is not something used in the rest of spoon or documented.  That's why I suggested introducing the `package` argument and then this validation would make more sense to me.

Reply:
Such package arg will be more consistent for sure - I can add it in scope of this PR. Still providing package over -e will be possible so we need such validation somewhere or users will hit strange behaviour described above.

Imho we should validate when --class-name, -e class, -e package and to be implemented --package-name are provided together. Question is: is there better place/way to do it? Should we limit -e args?

Reply:
Good point, users could still pass instrumentationArgs including package and could run into the issue you describe, even if there is a higher level package argument.  

This is probably a question for maintainers of spoon more than me, but IMO we can't protect users from everything they might pass into instrumentationArgs.  I'd lean towards just doing validation on `class-name` arg and `package` arg somewhere and not worry about validation on instrumentationArgs.